core/types/calls.cc

```diff
@@ -2132,6 +2132,14 @@ class Magic_selfNew : public IntrinsicMethod {
     }
 } Magic_selfNew;
 
+class Magic_nilForSafeNavigation : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, DispatchArgs args, DispatchResult &res) const override {
+        ENFORCE(args.args.size() == 1);
+        res.returnType = Types::nilClass();
+    }
+} Magic_nilForSafeNavigation;
```

Comment:
I don't think we need this, because you have already set the symbol's `resultType` in GlobalState.cc.

Reply:
Yup, everything seems to work without it! Will remove.